enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,340 @@
+---
+title: machine-config-node
+authors:
+  - "@cdoern"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@sinnykumari" # MCO
+  - "@yuqi-zhang" # MCO
+approvers:
+  - "@sinnykumari"
+  - "@yuqi-zhang"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@JoelSpeed"
+creation-date: 2023-10-05
+last-updated: 2023-10-20
+tracking-link:
+  - https://issues.redhat.com/browse/MCO-452
+see-also:
+replaces:
+superseded-by:
+---
+
+# Track MCO State Related to Upgrades in the MachineConfigNode type
+
+## Summary
+
+This enhancement describes how Nodes and their upgrade processes should be aggregated by the MCO into a user facing object. The goal here is to allow customers and the MCO team to decipher the our processes in a more verbose way, speeding up the debugging process and allowing for better customer engagement.
+
+## Motivation
+
+The MCO manages node upgrades but since we do not own the object or store much of their data in other ways, much of what occurs during an upgrade
+is simply a black box operation that we currently report as "Updating" or "Updated". Users can debug into a specific node or look into the node spec for some of this information
+but most of it simply lives in the MCO code rather than in data structures. We want to put these abstract "phases" of node operations as triggered by the MCO into a concrete data structure. 
+
+### User Stories
+
+* As a cluster admin, during upgrades I would like to monitor the progress of individual node upgrades so that, if one of the nodes sticks during an upgrade, I can easily see where the node got stuck and why.
+* As an OpenShift developer, I would like a place to observe the MCO's view of a Node so that I can easily identify if the MCO is related to an upgrade failure in CI.
+
+### Goals
+
+* Make a MachineConfigNode type that succinctly holds the upgrade data.
+* Have API load be as minimal as possible but augment the proper objects as needed.
+* aggregate as much MCO related data into easily accessible places as possible.
+
+### Non-Goals
+
+* Make existing/new API objects difficult to understand.
```

Comment:
nit: maybe we can just say something like "modify or remove existing API and status fields"